harness_utils/artifacts.py

```
@@ -0,0 +1,118 @@
+
+
+@unique
+class ArtifactType(Enum):
+    """
+    Describes different types of artifacts to be saved from test runs.
+    """
+
+    IMAGE_CONFIG = "image_config"
+    """
+    Meant to describe images displaying an applications settings.
+    For games this might be a screenshot of the in-game settings menu.
+    """
+
+    IMAGE_RESULTS = "image_results"
+    """
+    Meant to describe images displaying results of a benchmark.
+    For games with built in benchmarks, this might be a screenshot of an in-game benchmark results screen.
+    """
+
+    TEXT_CONFIG = "text_config"
+    """
+    Meant to describe text-based files which contain application settings.
+    For games this might be a .ini or .cfg file containing graphics settings.
+    """
+
+    TEXT_RESULTS = "text_results"
+    """
+    Meant to describe text-based files which contain results of a benchmark.
+    For games with built in benchmarks, this might be a .txt or .xml file containing results from an in-game benchmark.
+    """
+
+
+_IMAGE_ARTIFACT_TYPES = (ArtifactType.IMAGE_CONFIG, ArtifactType.IMAGE_RESULTS)
+
+
+@dataclass
+class Artifact:
+    """
+    Describes an artifact captured by the ArtifactManager.
+    """
+    filename: str
+    type: ArtifactType
+    description: str
+
+    def as_dict(self) -> dict:
+        """
+        Returns the Artifact object as a dictionary. Converts its type to a string value.
+        """
+        d = self.__dict__.copy()
+        d["type"] = d["type"].value
+        return d
+
+
+class ArtifactManager:
+    """
+    Used to manage artifacts captured during a test run, either by coping files or taking screenshots.
+    The manager maintains a list of artifacts it has captured and can produce a manifest file listing them.
+    """
+    def __init__(self, output_path: str | os.PathLike) -> None:
+        self.output_path = Path(output_path)
+        self.artifacts: list[Artifact] = []
+
+        self.output_path.mkdir(parents=True, exist_ok=True)
```

Can we try/catch here and raise whichever exception was given when trying to mkdir. 

Ideally if there is a permission problem, or creating dir problem we can fail fast.

---

Yeah you are right